### PR TITLE
Ub fix bandersnatch provider

### PR DESCRIPTION
**Pull request by pcseanmckay**
Updated bandersnatch provider to better parse the provider.conf config file.

### CODE REVIEW

**Review by codrcodz — CHANGES_REQUESTED** [review]
Same feedback as #6 

- Avoid calling external binaries when provisions already exist in bash for the thing you are trying to do
- Should we use sed instead of appending to the end of the file, or does location of not matter? What happens if two lines for the same parameter get dropped in the same file? does it accept the final one?

**Comment by pcseanmckay** [comment]
See my relevant comments in PR #6.  Once again if you agree, then I will make the necessary changes.

File: providers.d/bandersnatch/provider_wrapper.sh [hunk 1]
```diff
@@ -12,16 +12,35 @@ bashelliteProviderWrapperBandersnatch() {
...
+    if [[ ${line} =~ ^directory[[:blank:]]*[=][[:blank:]]*[[:alnum:]/_-.]+ ]]; then
+      if [ "${line_counter}" = 0 ]; then
+        echo "directory = ${_r_mirror_tld}/${_n_repo_name}" >> ${tmp_config_file}
+      else
+        utilMsg WARN "$(utilTime)" "Duplicate directory parameter found.  Skipping line..."
+      fi
+    else
+      echo "${line}" >> ${tmp_config_file}
+    fi
+    line_counter=$(($line_counter + 1))
```
Comment (codrcodz) [inline providers.d/bandersnatch/provider_wrapper.sh]: This line needs to be inside the while loop near your `echo "directory` line. Right now, if the first line in the config file is something besides `^directory =`, your line_counter variable will get incremented, and your line_counter test will always hit your `else` conditional for each subsequent line.

**Review by codrcodz — CHANGES_REQUESTED** [review]
Correct logic issue.

**Review by codrcodz — APPROVED** [review]

**Pull request merged**